server/client.go

```diff
@@ -831,6 +832,22 @@ func (c *client) sendPing() {
 	c.sendProto([]byte("PING\r\n"), true)
 }
 
+// Generates the INFO to be sent to the client with the client ID included.
+// info arg will be copied since passed by value.
+// Assume lock is held.
+func (c *client) generateClientInfoJSON(info Info) []byte {
+	info.CID = c.cid
+	// Generate the info json
+	b, err := json.Marshal(info)
+	if err != nil {
```

Comment:
My take on this is either we should ignore error or panic. That is, there is no reason Marshal() should return an error (we control what goes into the structure), but if it does, then something is really wrong and we should fix the bug.

Reply:
Not a fan of panic and killing the server, but maybe. s.Fatalf essentially did that, could change it back to that.

Reply:
That's fine. Like I said, not sure it will ever return an error, so...